### PR TITLE
gpaddmirror: add test coverage

### PR DESCRIPTION
**Pull request by larham**
We are porting Tinc test_gpaddmirrors.py to Behave.  We have added unit tests as needed and commented out the corresponding tests in test_gaddmirrors.py.

### CODE REVIEW

**Comment by ashwinstar** [comment]
Why "commented out the corresponding tests in test_gaddmirrors.py", delete it ?

**Comment by larham** [comment]
@ashwinstar Good point about deleting.  The plan is to delete the entire file when finished with followup work.  We felt like the comments helped us know where the unfinished work began, since there is a lot of other code in that file.

Instead, we'll delete the unwanted tests and add a big comment bar to indicate "next to be ported".

File: gpMgmt/bin/gppylib/test/unit/test_unit_gpaddmirrors.py [hunk 1]
```diff
@@ -22,33 +28,39 @@ def setUp(self):
...
-        with self.assertRaises(Exception):
-            self.subject.validate_heap_checksums(self.subject(None), self.gparrayMock)
-
-    def test_validate_heap_checksum_succeds_if_cluster_consistent(self):
+    def test_validate_heap_checksum_succeeds_if_cluster_consistent(self):
         self.mock_heap_checksum.return_value.get_segments_checksum_settings.return_value = ([1], [1])
         self.mock_heap_checksum.return_value.are_segments_consistent.return_value = True
         self.mock_heap_checksum.return_value.check_segment_consistency.return_value = ([2], [], 1)
-        self.subject.validate_heap_checksums(self.subject(None), self.gparrayMock)
+        self.subject.validate_heap_checksums(self.gparrayMock)
```
Comment (Chibin) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gpaddmirrors.py]: 👍

File: gpMgmt/bin/gppylib/test/unit/test_unit_gpaddmirrors.py [hunk 2]
```diff
@@ -58,17 +70,62 @@ def test_run_calls_validate_heap_checksum(self):
...
+    @patch('sys.stdout', new_callable=StringIO.StringIO)
+    def test_option_version(self, mock_stdout):
+        sys.argv = ['gpaddmirrors', '--version']
+        with self.assertRaises(SystemExit) as cm:
+            options, _ = self.parser.parse_args()
+
+        self.assertIn("gpaddmirrors version $Revision$", mock_stdout.getvalue())
+        self.assertEquals(cm.exception.code, 0)
+
+    def test_generated_file_contains_default_port_offsets(self):
```
Comment (Chibin) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gpaddmirrors.py]: Sweet, glad to see this here instead of tinc

**Pull request merged**